booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,659 @@
+        issues.add("Port of Load/Place of Receipt values must be provided");
+      }
+      if(!"SD".equals(receiptTypeAtOrigin) && ielNode != null) {
+        issues.add("Container intermediate export stop-off location should not be provided");
+      }
+
+      return issues;
+    });
+
+
+  private static final JsonContentCheck REQUESTED_CHANGES_PRESENCE = JsonAttribute.customValidator(
+    "Requested changes must be present for the selected Booking Status ",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus").asText("");
+      var issues = new LinkedHashSet<String>();
+      if (PENDING_CHANGES_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        var requestedChanges = body.path("requestedChanges");
+        if (requestedChanges == null) {
+          issues.add("requestedChanges is missing in allowed booking states %s".formatted(PENDING_CHANGES_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck REASON_PRESENCE = JsonAttribute.customValidator(
+    "Requested changes must be present for the selected Booking Status ",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus").asText("");
+      var issues = new LinkedHashSet<String>();
+      if (REASON_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        var reason = body.path("reason");
+        if (reason == null) {
+          issues.add("reason is missing in the Booking States %s".formatted(REASON_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck CHECK_CONFIRMED_BOOKING_FIELDS = JsonAttribute.customValidator(
+    "check confirmed booking fields availability",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var bookingStatus = body.path("bookingStatus").asText("");
+      if (CONFIRMED_BOOKING_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        if (body.path("confirmedEquipments") == null) {
+          issues.add("confirmed Equipments for confirmed booking is not present");
+        }
+        if (body.path("termsAndConditions") == null) {
+          issues.add("confirmed Equipments for confirmed booking is not present");
+        }
+        if (body.path("transportPlan") == null) {
+          issues.add("transportPlan for confirmed booking is not present");
+        }
+        if (body.path("clauses") == null) {
+          issues.add("clauses for confirmed booking is not present");
+        }
+        if (body.path("charges") == null) {
+          issues.add("charges for confirmed booking is not present");
+        }
+        if (body.path("shipmentCutOffTimes") == null) {
+          issues.add("shipmentCutOffTimes for confirmed booking is not present");
+        }
+        if (body.path("advanceManifestFilings") == null) {
+          issues.add("advanceManifestFilings for confirmed booking is not present");
+        }
+      }
```

Comment:
The `.path` cannot return null, which makes many of these `if` statements always be false.